osquery/tables/system/linux/processes.cpp

```diff
@@ -56,6 +56,26 @@ inline std::string readProcLink(const std::string& attr, const std::string& pid)
   return result;
 }
 
+Status getPidsForContext(QueryContext& context, std::set<std::string>& pids) {
+  auto s = osquery::procProcesses(pids);
+  if (!s.ok()) {
+    return s;
+  }
+
+  if (context.constraints.count("pid") > 0 &&
+      context.constraints["pid"].exists(EQUALS) &&
+      context.constraints.at("pid").getAll(EQUALS).size() < pids.size()) {
+    pids.clear();
+    for (const auto& pid :
+         context.constraints["pid"].getAll<std::string>(EQUALS)) {
+      if (isDirectory("/proc/" + pid).ok()) {
```

Comment:
Don't you already know if the pid was in the `pids` set? What if you construct a new set if the constraint exists and move matching pids over? That will be considerably less work than `stat`ing the directory.